### ISSUE TITLE
A multitude of games stop working with glibc > 2.19

### ISSUE THREAD

**Issue by damianatorrpm** [flathub/com.valvesoftware.Steam#595]
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1092
https://steamcommunity.com/app/282070/discussions/0/610573751159186268/?ctp=4#c530647080133257413
https://wiki.archlinux.org/index.php/Steam/Game-specific_troubleshooting#Infinite_loading

Flatpak latest but it's kinda irrelevant since this happens also on system steam if you're not on Ubuntu.
The steam flatpak should ship it's own glibc or preload the fix.so

**Comment by nanonyme** [comment]
Shipping glibc inside the app is really a no-go. Need more conscice info about workaround, the forum is quite hard to read through

**Comment by nanonyme** [comment]
The workaround looks so full of hack that I would prefer it getting applied *only* for this game through shared-library-guard. Need to look into this. I don't have game myself.

**Comment by nanonyme** [comment]
Is there a bug filed at https://github.com/ValveSoftware/steam-for-linux on this so Valve to make sure Valve is aware of this rather esoteric issue? How many games are affected?

**Comment by damianatorrpm** [comment]
Yes, there are bugs filled and open since 2015. 
From the ones I own and tested 7 games stay stuck at loading screen forever.

All of these state to be supported with Ubuntu 12.04 which has glibc older than 2.19,
where it works.

Valve doesn't care at all since the publishers supported OS'es are not used i.e Ubuntu 12.04

**Comment by nanonyme** [comment]
Which games are those?

**Comment by damianatorrpm** [comment]
Anomaly 2
Anomaly Defenders
Anomaly Korea
Anomaly Warzone Earth
Anomaly Warzone Earth Mobile Campaign
This War of Mine
And I'm sure there was antoher one...

**Comment by nanonyme** [comment]
Note that you probably want to put the library into .var/app/com.valvesoftware.Steam, then flatpak run --env=LD_PRELOAD=theFixLibrary com.valvesoftware.Steam if you want to use it as a workaround.

**Comment by damianatorrpm** [comment]
@nanonyme I don't think that works, since steam when launcing games seems to override LD_PRELOAD again (?!)
unless flatpak run forces this env variable to be persistent but this doesn't work always e.g chrome.
